inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/multiple/IcebergSingleStreamWriter.java

```diff
@@ -84,11 +135,60 @@ public void prepareSnapshotPreBarrier(long checkpointId) throws Exception {
 
     @Override
     public void processElement(T value) throws Exception {
-        writer.write(value);
+        try {
+            writer.write(value);
+        } catch (Exception e) {
+            if (multipleSink) {
+                throw e;
+            }
+
+            LOGGER.error(String.format("write error, raw data: %s", value), e);
+            if (!dirtyOptions.ignoreDirty()) {
+                throw e;
+            }
+            if (dirtySink != null) {
+                DirtyData.Builder<Object> builder = DirtyData.builder();
+                try {
+                    builder.setData(value)
+                            .setLabels(dirtyOptions.getLabels())
+                            .setLogTag(dirtyOptions.getLogTag())
+                            .setIdentifier(dirtyOptions.getIdentifier())
+                            .setRowType(flinkRowType)
+                            .setDirtyMessage(e.getMessage());
+                    dirtySink.invoke(builder.build());
+                    if (metricData != null) {
+                        metricData.invokeDirtyWithEstimate(value);
+                    }
+                } catch (Exception ex) {
+                    if (!dirtyOptions.ignoreSideOutputErrors()) {
+                        throw new RuntimeException(ex);
+                    }
+                    LOGGER.warn("Dirty sink failed", ex);
+                }
+            }
+        }
+        if (metricData != null) {
+            metricData.invokeWithEstimate(value == null ? "" : value);
+        }
     }
 
     @Override
     public void initializeState(FunctionInitializationContext context) throws Exception {
+        // init metric state
+        if(multipleSink){
+            return;
+        }
+        if (this.inlongMetric != null) {
+            this.metricStateListState = context.getOperatorStateStore().getUnionListState(
+                    new ListStateDescriptor<>(
+                            String.format("Iceberg(%s)-" + INLONG_METRIC_STATE_NAME, fullTableName),
```

Comment:
just use INLONG_METRIC_STATE_NAME. Because It will cause metric restore error for old task

Reply:
Fixed